Building.md

```diff
@@ -166,16 +188,3 @@ build system.)
 
 Specifically some advanced techniques to obtain performance deltas for the
 GC can be found in `rts/Benchmarking.md`.
-
-## Updating Haskell Packages
-
-When the `.cabal` file of a Haskell package is changed you need to make sure the
-corresponding `.nix` file (stored in `nix/generated/`) is kept in sync with it. These files are automatically generated; run
-```
-nix-shell nix/generate.nix
-```
-to update.
-
-Don't worry if you forget to update the `default.nix` file, the CI job
-`check-generated` checks if these files are in sync and fail with a diff if
-they aren't.
```

Comment:
Note to reviewers: we now use the nixpkgs function `callCabal2nix` to generate the derivations for our Haskell packages which means we no longer need to pre-generate the .nix files. This does use IFD (Import From Derivation) but since we no longer use Hydra to build things we don't care about that.